packages/react/src/lib/factories.ts

```diff
@@ -263,7 +263,14 @@ function createShorthandFromValue({
 
     if (valIsReactElement) {
       // use the key from React Element
-      props.key = (value as React.ReactElement).key
+      const elementKey = (value as React.ReactElement).key
+      // <div /> - key is not passed as will be `null`
+      // <div key={null} /> - key is passed as `null` and will be stringified
+      const isNullKey = elementKey === 'null' || elementKey === null
```

Comment:
afraid that there may be legit case when use would want to provide `null` as a string key for an element

Reply:
Valid point 👍 
Fixed